### DISCUSSION TITLE
Are there any examples of recording sound captured by the I2S mic to wav or mp3 file and store it on SD card?

### DISCUSSION THREAD

**Discussion by wimverlindensa** [pschatzmann/arduino-audio-tools#188] [Q&A] [answered]
Hi are there any examples of recording sound captured by the I2S mic to a wav and  even better to a  mp3 file and store it on SD card?

I found a good working example to playback  an mp3 file on the SD card by I2S, but not the opposite!

Thanks!

**Comment by pschatzmann** [comment]
Did you read the [Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki)? There is a chapter about encoding and decoding.

**Reply by wimverlindensa** [reply]
I checked it but did not find an example for I2S ( mic) to MP3. 
Hopefully more examples will be added in the future.

**Comment by pschatzmann** [comment, marked as answer]
Unfortunately I can't make ever possible combination as example: Just read the Wiki about encoding and you should be able to encode any supported file format from any source to any destination...

**Comment by pschatzmann** [comment]
I finally added some [file based recording example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-audiokit-sd-audiokit/streams-audiokit-sd-audiokit.ino). The conversion to the needed format should be simple...

**Comment by wimverlindensa** [comment]
Great!

 

Van: Phil Schatzmann ***@***.***> 
Verzonden: donderdag 1 september 2022 20:34
Aan: pschatzmann/arduino-audio-tools
***@***.***>
CC: wimverlindensa ***@***.***>; Author
***@***.***>
Onderwerp: Re: [pschatzmann/arduino-audio-tools] Are there any examples of
recording sound captured by the I2S mic to wav or mp3 file and store it on
SD card? (Discussion #188)

 

I finally added some file based recording example
<https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examp
les-audiokit/streams-audiokit-sd-audiokit/streams-audiokit-sd-audiokit.ino>
. The conversion to the needed format should be simple...

-
Reply to this email directly, view it on GitHub
<https://github.com/pschatzmann/arduino-audio-tools/discussions/188#discussi
oncomment-3529661> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AHHQX7G5TSCTEGGZEE5C2V3V4
DZLDANCNFSM5UY7IVUA> .
You are receiving this because you authored the thread.Message ID:
***@***.***
b.com>

**Comment by wimverlindensa** [comment]
yes, i did not find an example of recording i2s mic in mp3 on sd card. 


Van: "Phil Schatzmann" ***@***.***> 
Aan: "pschatzmann/arduino-audio-tools" ***@***.***> 
Cc: "wim verlinden" ***@***.***>, "Author" ***@***.***> 
Verzonden: Zondag 1 mei 2022 01:18:14 
Onderwerp: Re: [pschatzmann/arduino-audio-tools] Are there any examples of recording sound captured by the I2S mic to wav or mp3 file and store it on SD card? (Discussion #188) 





Did you read the [ https://github.com/pschatzmann/arduino-audio-tools/wiki | Wiki ] ?
